### ISSUE TITLE
[ban-types] false negative when banned type name is used locally

### ISSUE THREAD

**Issue by tpluscode** [typescript-eslint/typescript-eslint#2851]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

https://github.com/tpluscode/eslint-ban-types

**Expected Result**

No lint errors, because the type `Function` is not the built-in type but a local interface

**Actual Result**

```
/Users/tomaszpluskiewicz/projects/temp/eslint/index.ts
  5:24  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
```

**Additional Info**

n/a

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.9.0` |
| `@typescript-eslint/parser`        | `4.9.0` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `14.13.1` |

**Comment by bradzacher** [comment]
The rule just bans the type name. It does not do anything smarter than that.
It does not care if it's local, imported, global or anything in between. It just bans the name.
So this is working as intended.

----

As an aside - you should avoid shadowing global type names.
Your code will be very confusing to understand if you do this.
Why? If you ever delete or rename your shadow implementation but forget to rename a usage - your code will either silently continue to work, or TS will start to error in weird ways.
